### ISSUE TITLE
Register new user via email fails with "error occured" alert

### ISSUE THREAD

**Issue by petervojtek** [PowerlineApp/powerline-mobile#60]
in android, I fill in all the fields, click finish registration, and "error occured" alert appers.
this happens on current 'develop' branch, on commit 707e65cb14b9300bb.

**Comment by luckydevstar** [comment]
It is plugin error. If you figure out all plugin correctly, that error will not occure.

**Comment by petervojtek** [comment]
@igaponov, when I send new user registration request, I receive http response 400, and e.g. `"property":"zip","message":"This value should not be blank."` despite I filled in the zip code in request:

```
Request URL:https://api-dev.powerli.ne/api/secure/registration
Request Method:POST
Status Code:400 Bad Request
Remote Address:52.5.45.80:443

request body:
first_name=Peter2&last_name=Peter2&username=Peter2&password=Peter2&confirm=Peter2&address1=San+Di&city=San+Diego&state=CA&country=US&zip=92093&email=peter3%40test&email_confirm=peter3%40test&phone=12345&birth=01%2F01%2F1977&address2=123

response:
{"errors":[{"property":"username","message":"This value should not be blank."},
{"property":"password","message":"This value should not be blank."},
{"property":"firstName","message":"This value should not be blank."},
{"property":"lastName","message":"This value should not be blank."},
{"property":"email","message":"This value is not a valid email address."},
{"property":"zip","message":"This value should not be blank."}]}
```

Also `first_name`, `last_name` and `email` receive validation errors (value is blank) despite they are not blank.

**Comment by igaponov** [comment]
```
http POST localhost:8080/app_dev.php/api/secure/registration first_name=Peter last_name=Peter username=Peter password=Peter confirm=Peter address1=San+Di city=San+Diego state=CA country=US zip=92093 email=peter2@test.com email_confirm=peter2@test.com phone=123456 birth=01-01-1977 address2=123 --form
HTTP/1.1 200 OK
Access-Control-Allow-Headers: accept, origin, x-requested-with, Authorization, content-type, Token
Access-Control-Allow-Methods: GET, POST, PUT, DELETE, OPTIONS, PATCH
Access-Control-Allow-Origin: *
Cache-Control: no-cache
Connection: keep-alive
Content-Type: application/json
Date: Tue, 21 Jun 2016 11:03:10 GMT
Server: nginx/1.4.6 (Ubuntu)
Transfer-Encoding: chunked
X-Debug-Token: 6fa2d2
X-Powered-By: PHP/5.5.9-1ubuntu4.17

{
    "id": 2, 
    "is_registration_complete": true, 
    "token": "1pwn0wby2fa8ksg4csck8woswsk0wkkso8sks48ck0k4gwkwg4", 
    "username": "Peter"
}
```

I'm not sure how do you send your data. If you put it to body, you must add the right content type (`multipart/form-data` in your case). But it's better to send json with a type `application/json`.

**Comment by petervojtek** [comment]
thanks for helpful reply @igaponov , I confirm that when I switched to json, registration works properly.

**Comment by petervojtek** [comment]
fixed in commit [55a5e85df238](https://github.com/PowerlineApp/powerline-mobile/commit/55a5e85df238d6b1c78224ee47db299825460c07)
